google-cloud-firestore/src/main/java/com/google/cloud/firestore/BulkWriter.java

```diff
@@ -515,70 +756,117 @@ private void verifyNotClosed() {
     }
   }
 
+  /**
+   * Attaches a listener that is run every time a BulkWriter operation successfully completes.
+   *
+   * <p>For example, see the sample code: <code>
+   *   BulkWriter bulkWriter = firestore.bulkWriter();
+   *   bulkWriter.addWriteResultListener(
+   *         new WriteResultCallback() {
+   *           public void onResult(DocumentReference documentReference, WriteResult result) {
+   *             System.out.println(
+   *                 "Successfully executed write on document: "
+   *                     + documentReference
+   *                     + " at: "
+   *                     + result.getUpdateTime());
+   *           }
+   *         });
```

Comment:
You should probably just use Lamdba syntax here.

Reply:
done.